libsnark/zk_proof_systems/ppzksnark/r1cs_gg_ppzksnark/r1cs_gg_ppzksnark.tcc

```diff
@@ -396,6 +387,26 @@ r1cs_gg_ppzksnark_keypair<ppT> r1cs_gg_ppzksnark_generator(const r1cs_gg_ppzksna
     return r1cs_gg_ppzksnark_keypair<ppT>(std::move(pk), std::move(vk));
 }
 
+template <typename ppT>
+r1cs_gg_ppzksnark_keypair<ppT> r1cs_gg_ppzksnark_generator(const r1cs_gg_ppzksnark_constraint_system<ppT> &r1cs)
+{
+    libff::enter_block("Call to r1cs_gg_ppzksnark_generator");
+
+    /* Generate secret randomness */
+    const libff::Fr<ppT> t = libff::Fr<ppT>::random_element();
+    const libff::Fr<ppT> alpha = libff::Fr<ppT>::random_element();
+    const libff::Fr<ppT> beta = libff::Fr<ppT>::random_element();
+    const libff::Fr<ppT> delta = libff::Fr<ppT>::random_element();
+    const libff::G1<ppT> g1_generator = libff::G1<ppT>::one();
+
+    r1cs_gg_ppzksnark_keypair<ppT> key_pair = r1cs_gg_ppzksnark_generator_from_secrets<ppT>(
+        r1cs, t, alpha, beta, delta, g1_generator);
```

Comment:
Why don't you pass the generator of `g2` as an argument to this function?

Reply:
Thanks.  Changed to add `g2_generator` as an argument.